sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/CosmosDBFactory.java

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.data.cosmos;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.spring.data.cosmos.common.MacAddress;
+import com.azure.spring.data.cosmos.common.PropertyLoader;
+import com.azure.spring.data.cosmos.common.TelemetrySender;
+import com.azure.spring.data.cosmos.config.CosmosDBConfig;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import javax.annotation.PostConstruct;
+import java.lang.reflect.Field;
+
+/**
+ * Factory class for CosmosDb to create client
+ */
+public class CosmosDBFactory {
```

Comment:
is this a public class? I wonder if it should be CosmosDBFactory or CosmosFactory. the same comment applies to all other public surface area, should we go with CosmosDB or Cosmos on the public surface. 

our brand name is Cosmos

Reply:
Yes, changing all CosmosDB to Cosmos in next commit on this PR.